pubsub/cloud-client/subscriber.py

```diff
@@ -461,6 +618,46 @@ def callback(message):
     # [END pubsub_subscriber_error_listener]
 
 
+def receive_messages_with_delivery_attempts(
+    project_id, subscription_name, timeout=None
+):
+    # [START  pubsub_dead_letter_delivery_attempt]
+    from google.cloud import pubsub_v1
+
+    # TODO project_id        = "Your Google Cloud Project ID"
+    # TODO subscription_name = "Your Pubsub subscription name"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name
+    )
+
+    def callback(message):
+        print("Received message: {}".format(message))
+        print("With delivery attempts: {}".format(message.delivery_attempt))
+        message.ack()
+
+    streaming_pull_future = subscriber.subscribe(
+        subscription_path, callback=callback
+    )
+    print("Listening for messages on {}..\n".format(subscription_path))
+
+    # Wrap subscriber in a 'with' block to automatically call close() when done.
+    with subscriber:
+        # When `timeout` is not set, result() will block indefinitely,
+        # unless an exception is encountered first.
+        try:
+            streaming_pull_future.result(timeout=timeout)
+        except Exception as e:
```

Comment:
Catching `Exception` here is too broad - can you either narrow this down, or else remove try/catch and just let the exception propagate?

Reply:
Done.